### PR TITLE
[SYCL-MLIR][FIX] Make sure that the global strings goes to the proper module.

### PR DESCRIPTION
**Pull request by arnamoy10**

### CODE REVIEW

File: polygeist/tools/cgeist/Lib/clang-mlir.cc [hunk 1]
```diff
@@ -2235,12 +2235,18 @@ MLIRASTConsumer::getOrCreateGlobal(const ValueDecl &VD, std::string Prefix,
...
   using namespace mlir;
   // Create the global at the entry of the module.
   if (llvmStringGlobals.find(value.str()) == llvmStringGlobals.end()) {
     OpBuilder::InsertionGuard insertGuard(builder);
-    builder.setInsertionPointToStart(module->getBody());
+
+    if (funcContext == FunctionContext::SYCLDevice)
+      builder.setInsertionPointToStart(getDeviceModule(*module).getBody());
+    else
+      builder.setInsertionPointToStart(module->getBody());
```
Comment (whitneywhtsang) [inline polygeist/tools/cgeist/Lib/clang-mlir.cc:2248]:
```suggestion
    else {
      assert(FuncContext == FunctionContext::Host);
      builder.setInsertionPointToStart(module->getBody());
    }
```

File: polygeist/tools/cgeist/Test/Verification/sycl/global_string.cpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
+//RUN: clang++ -fsycl -fsycl-device-only -emit-mlir %s -S -o - 2>&1| FileCheck %s --check-prefix=MLIR
```
Comment (whitneywhtsang) [inline polygeist/tools/cgeist/Test/Verification/sycl/global_string.cpp:1]:
```suggestion
// RUN: clang++ -fsycl -fsycl-device-only -emit-mlir %s -S -o - 2>&1| FileCheck %s --check-prefix=MLIR
```
Can you please format the test case?

**Review by victor-eds — APPROVED** [review]
LGTM after Whitney's suggestions

File: polygeist/tools/cgeist/Test/Verification/sycl/global_string.cpp (hunk 2 above)
Comment (whitneywhtsang) [reply polygeist/tools/cgeist/Test/Verification/sycl/global_string.cpp]: Can you please format the test case in general?

**Review by victor-eds — CHANGES_REQUESTED** [review]
I'd say this needs rebasing

File: polygeist/tools/cgeist/Lib/clang-mlir.h [hunk 3]
```diff
@@ -190,7 +190,8 @@ class MLIRASTConsumer : public clang::ASTConsumer {
   /// given name, creating the string if necessary.
   mlir::Value GetOrCreateGlobalLLVMString(mlir::Location Loc,
                                           mlir::OpBuilder &Builder,
-                                          clang::StringRef Value);
+                                          clang::StringRef Value,
+                                          FunctionContext funcContext);
```
Comment (whitneywhtsang) [inline polygeist/tools/cgeist/Lib/clang-mlir.h:194]:
```suggestion
                                          FunctionContext FuncContext);
```

File: polygeist/tools/cgeist/Test/Verification/sycl/global_string.cpp [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  auto q = sycl::queue(d); 
+  {
+	  q.submit([&] (sycl::handler& cgh) {
+       cgh.single_task<class printkernel>([=] {
+          // Test that the string is generated in the gpu module
+          // MLIR: gpu.module @device_functions {
+          // MLIR-NEXT: llvm.mlir.global internal constant @str0("Hello\00") {addr_space = 0 : i32}
+          char str1[10] = "Hello";
+        });
+    });
```
Comment (whitneywhtsang) [inline polygeist/tools/cgeist/Test/Verification/sycl/global_string.cpp:14]:
```suggestion
    q.submit([&] (sycl::handler& cgh) {
       cgh.single_task<class printkernel>([=] {
          // Test that the string is generated in the gpu module
          // MLIR: gpu.module @device_functions {
          // MLIR-NEXT: llvm.mlir.global internal constant @str0("Hello\00") {addr_space = 0 : i32}
          char str1[10] = "Hello";
       });
    });
```

**Review by whitneywhtsang — APPROVED** [review]

File: polygeist/tools/cgeist/Test/Verification/sycl/global_string.cpp (hunk 2 above)
Comment (arnamoy10) [reply polygeist/tools/cgeist/Test/Verification/sycl/global_string.cpp]: Done

**Pull request closed without merging**